### DISCUSSION TITLE
Crashes on Split screen and long Jupyter notebook

### DISCUSSION THREAD

**Discussion by teddyhla** [posit-dev/positron#3963] [Feedback]
Thanks so much for all your efforts bringing Positron.

I have used it for about ten days and I really like how easily python envs are detected, and plots are shown. 

My main issue at the moment is it struggles with *split editor* tab especially on long jupyter notebook (100-200 lines, 50 odd code cells with 5-10 plots). It either freezes or crashes completely.

Please see attached image.

For specification, 

Apple M2 Pro 2023 
32GB Memory
macOS Ventura 13.4.1

Positron Version: 2024.07.0 (Universal) build 14
Code - OSS Version: 1.90.0
Commit: b750305d0e9079e1a6d4e0b1d78daa70964b0cb8
Date: 2024-07-04T04:17:07.822Z
Electron: 29.4.0
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Darwin arm64 22.5.0 

Running a conda virtual environment 

Working on a tabular data 10k rows or so CSV file size 1.8 to 5 MB.

![reprex](https://github.com/posit-dev/positron/assets/60614830/f862bed1-89ec-4f12-978d-8a6e85733b9f)

**Comment by juliasilge** [comment]
I just spent a little time with two long-ish Jupyter notebooks in split screen and wasn't able to produce a crash. Do you notice the crash happening when you are doing anything in particular, like adding a new cell or typing?

**Reply by teddyhla** [reply]
Hi Julia. Thanks. It's when adding new cells and scrolling up and down. It looks like it's trying to render but got stuck. As you may see in the pic (sorry if small), cell 36 skipped to 42 without showing the contents in the middle (plain text/code cells/markdown). I managed to run the same notebook in VSCode and was fine. 

I will try to replicate the problem again.

**Reply by seeM** [reply]
@teddyhla the entire window crashing unfortunately makes it a bit harder for us to figure out what's going on. Are you able to reliably reproduce this with a few steps, and perhaps with a smaller notebook or a subset of the data? Alternatively, are you able to share these notebooks/datasets for us to try and repro on our side?

You mentioned the cell skipping from 36 to 42. How did you run the cells leading up to that, was it via the "Run All" button or did you run individual cells? That might be a clue.

You should also be able to dig up the kernel logs after the crash, but it is a bit messy! Start by running the following command (I've only tested this on macOS with the standard `ls` builtin):

```sh
ls -dt "$TMPDIR"kernel-*/ | head -n 5
```

For example, I see:

```
/var/folders/_r/pm17dpfn7lx5mf3wtd96kw3m0000gn/T/kernel-ljDkRU/
/var/folders/_r/pm17dpfn7lx5mf3wtd96kw3m0000gn/T/kernel-tQNp8f/
/var/folders/_r/pm17dpfn7lx5mf3wtd96kw3m0000gn/T/kernel-mYMOel/
/var/folders/_r/pm17dpfn7lx5mf3wtd96kw3m0000gn/T/kernel-Pef3bQ/
/var/folders/_r/pm17dpfn7lx5mf3wtd96kw3m0000gn/T/kernel-5FuObf/
```

Each of these corresponds to a separate kernel starting in Positron, and contains a `kernel.log` file. First double-check which of those corresponds with your notebook. One way to do that is to search the file for `execute_request` messages and check whether they match code you executed.

If the log file looks a bit empty, you may have to change the kernel log level. Search settings for `python.languageServerLogLevel` and set that to `debug`. That will have to be done _before_ the notebook kernel starts (and eventually crashes).

**Reply by Heseraj** [reply]
<img width="3840" height="2088" alt="image" src="https://github.com/user-attachments/assets/1d2c71ab-0638-4a27-ae7f-9873be2d4a16" />
this has been popping up on my computer recently. 
Positron not Responding. Wonder if anyone else has had the same issue?

**Reply by juliasilge** [reply]
Can you say more about when you are observing this? Is it after a long-running session, or anytime you open Positron?

**Reply by Heseraj** [reply]
Actually, the majority of cases, it happens when I want to see the output of a function call that might have 1000+ lines. In this specific example, it happened when I was playing around with an NLP lecture. 
<img width="2077" height="469" alt="image" src="https://github.com/user-attachments/assets/f8091262-6124-4de5-9d1e-a7e52afb3b96" />
It always happens when train_test_int or test_text_in is printed on console. I just ran to it again. 
In order for it to not happens, I print a limited number of lines. 
Also, in order to resolve it, I waited for the program to respond but nothing happened. So normally close the IDE and re-open it. 
![Uploading image.png…]()

**Reply by Heseraj** [reply]
<img width="3840" height="2088" alt="image" src="https://github.com/user-attachments/assets/8cfe98b4-50b5-48bd-9d4d-58b08cbe7546" />

**Reply by juliasilge** [reply]
So you're printing something in the console that generates over 1000+ lines of output? Do I understand that right?

**Reply by Heseraj** [reply]
Yes.

**Reply by juliasilge** [reply]
I wonder if you are running into something related to https://github.com/posit-dev/positron/issues/9852. 🤔 

I just tried running something like this:

```python
output = "\n".join(f"Line {i}: This is output line number {i}" for i in range(1, 1051))
print(output)
```

And I did not observe any problems. @Heseraj, does simple printing like that cause Positron to hang for you, in a fresh session? Or is something more complicated going on?
